server/src/instant/storage/s3.clj

```diff
@@ -90,11 +94,32 @@
 ;; Instant <> S3 integration
 ;; ----------------------
 
-(defn upload-file-to-s3 [{:keys [app-id location-id] :as ctx} file]
-  (when (not (instance? java.io.InputStream file))
+(def ^Tika tika (Tika.))
+
+(defn requires-tika? [{:keys [content-type app-id]}]
+  (and
+   (not content-type)
+   (contains? (flags/flag :tika-enabled-apps) app-id)))
+
+(defn deduce-content-type [{:keys [content-type] :as ctx} file]
+  (if (requires-tika? ctx)
+    (.detect tika ^TikaInputStream file)
+    content-type))
+
+(defn get-file-stream [ctx file]
```

Comment:
```suggestion
(defn get-file-stream ^InputStrem [ctx ^InputStream file]
```

With this change, the type hints act as extra documentation. Then you can get rid of the inline type hint on line 111 and the type hint on line 117.